src/partr.c

```diff
@@ -476,8 +476,7 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q)
                     JL_UV_UNLOCK();
                     // optimization: check again first if we may have work to do
                     if (!may_sleep(ptls)) {
-                        if (ptls->sleep_check_state != not_sleeping)
-                            jl_atomic_store(&ptls->sleep_check_state, not_sleeping); // let other threads know they don't need to wake us
+                        assert(ptls->sleep_check_state == not_sleeping);
```

Comment:
Should we just make every access of `sleep_check_state` atomic as a matter of course?

```suggestion
                        assert(jl_atomic_load_relaxed(&ptls->sleep_check_state) == not_sleeping);
```

Reply:
I agree. The change to use `assert` also makes sense (since that's currently the implementation of `may_sleep`), but assumes that we don't alter the implementation of `may_sleep` to possibly have disjoint conditions that we want to check (i.e. the original partr code had a global `sleep_check_state`, so that bulk operations were also attempted).